File: src/main/java/model/ModifiableArena.java
package model;

public class ModifiableArena extends Arena {

	public void addSnake(ModifiableSnake snake) {
		this.snakes.add(snake);
	}

	public void move() {
		printResultsIfNeeded();
		for (ModifiableSnake snake : snakes) {
			snake.move();
		}
		round++;
	}

	public void removeFood(Coordinate nextCoordinate) {
		removeFoodFromCollection(nextCoordinate);
		generateNewFood();
	}

}

File: src/main/kotlin/co/electric/snake/model/ModifiableArena.kt
package co.electric.snake.model

class ModifiableArena : Arena() {

    fun addSnake(snake: ModifiableSnake) {
        snakes.add(snake)
    }

    fun move() {
        printResultsIfNeeded()
        snakes.stream().forEach(ModifiableSnake::move)
        round++
    }

    fun removeFood(nextCoordinate: Coordinate) {
        removeFoodFromCollection(nextCoordinate)
        generateNewFood()
    }

}
